### ISSUE TITLE
Left click held down

### ISSUE THREAD

**Issue by eons44** [iiordanov/remote-desktop-clients#259]
Left click is held down.
This breaks most gui interactions.

To reproduce:
1. Install UserLAnd lxde or xfce environment.
2. Launch Samsung Dex
3. Open the session in bVNC

This issue is not present when using the Remote Desktop Manager app, from Google Play.
This issue also did not appear to be present when using the OpenBox desktop that UserLAnd employs by default.

I believe this is what S K and Oleg are referencing in their reviews of bVNC on Google Play, left 21 Mar 21.

Thank you and best of luck debugging this!

**Comment by iiordanov** [comment]
Hi there,

Could you kindly try some older relases from here: https://github.com/iiordanov/remote-desktop-clients/releases/ and let me know if any of them lack the issue?

Thanks!!
iordan

**Comment by iiordanov** [comment]
Hey, I just tested with my Samsung Note 9 updated to the latest available version of Android for it.

Installed UserLand.
Installed Ubuntu.
Ran Dex
Used v5.0.4_115041 to connect.

No input issues whatsoever.

**Comment by iiordanov** [comment]
Also installed XFCE

I could also not reproduce any input issues whatsoever with XFCE.

There have been no changes to the mouse support code recently, so it's likely this was a transient issue with some package userland installs. Try updating to latest and restarting XFCE and let me know if the issue persists.

**Comment by eons44** [comment]
Hey iordan,
Thanks for looking into this. Sorry, I haven't had the time to follow up. I'll do some more debugging if / when I get the chance.
However, I just had a random thought that might be related:
What android version are you on?

I saw this on Android 11, One UI 3.1, on a Note 20 Ultra.
Maybe this is an Android 11 thing?

If it's a transient issue, it's likely within one of the apps, as I recreated several containers trying to get this to work. 

p.s. updates do not seem to work on UserLAnd at the moment. Some of the upstream packages, like libc, appear to be compiled incorrectly. I assume you meant "update to the latest bVNC & UserLAnd apps"; I just figured it's worth noting that we probably shouldn't update xfce, etc. from within the container. Fingers crossed aarch64 gets more support in the coming years..

Thanks again,
eons

**Comment by iiordanov** [comment]
I was able to run apt update ; apt dist-upgrade inside the xfce container.

**Comment by eons44** [comment]
This is probably a divergence from the point of this issue but my curiosity's gotten the better of me.
Does `curl` work? That's what I was getting libc errors with. It works on a fresh image but not after updates.

**Comment by iiordanov** [comment]
Sadly no, there is a dependency missing, looks like.

I can't reproduce the mouse button held down issue at all though.

I have an external mouse and keyboard connected through a Lenovo hub, and I
am running DEX.

If you can give me instructions on how to reproduce please do so.

**Comment by eons44** [comment]
New requirement: Use Samsung dex on windows. 

I don't have a good standalone setup; so, I've been proving all this out on my laptop before I invest in anything.
However, I do have a usbc-to-hdmi adapter that, when used, appears to inhibit this issue. Unfortunately, the mouse is miscalibrated under that setup but that's fine.

Lemme know if that's still not reproducable.

bVNC @ v5.0.4
UserLAnd @ 2.7.3

**Comment by iiordanov** [comment]
So I have to install Windows and Samsung DEX on Windows and connect the
phone to Windows? That hardly seems like something people would do on a
regular basis - if you have a full desktop OS at hand, you would use DEX
only if there were indispensable Android software you need to use. It
hardly seems that bVNC is to blame if it works fine standalone, don't you
agree?

**Comment by eons44** [comment]
I think the reasons behind this setup are out of scope for this issue. Remote Desktop Manager does not exhibit this behavior; so, this is, afaik, a bug in bVNC. Additionally, windows being necessary to reproduce this is still in question.
Regardless, if this is not something that is going to be fixed, just document that and hopefully the people leaving 1 star reviews will read that. If we can confirm this is a windows issues, it's totally fine to say dex on windows isn't supported; that'll just mean less people use the app.

"Users" will always break whatever we build and find ways of using our tools for strange, seemingly unreasonable hacks. It's up to us, as developers, to decide how well we support them.

---

To address the "why would people ever do that" question, I'll throw out a few personal anecdotes:

1. The reason I am doing this now is, as I mentioned, to test the setup and make sure it's worth investing in.

2. The reason I'm even considering this approach in the first place is that I'll be able to reduce the weight I carry while traveling if I can manage my remote servers from my phone. Maybe I'll need to borrow someone's laptop to do some emergency maintence; who knows what odd situation I'll find myself in where I need dex on windows?

3. The other reason I'm trying this approach is that my machines break all the time and for no good reason. My phone is not likely to become less important. So, if I can find a way to do all my work on my phone and take regular, automated backups of it, I don't need to worry about running clonezilla ever week or whatever. If that is a success, I'd be able to wipe my laptops and just install dex on them. This isn't possible yet (e.g. adobe suite) but I think there's a workflow that incorporates android.

4. In ancient times, before samsung dex and UserLAnd (~5 years ago), I worked at a university that had need of my web development skills (not what I was hired for). Through all the university bureaucracy, I wasn't given the ability to run a webserver on the computer they provided me nor was I wable to remote access my computer at home (and maybe my laptop was broken or something). Anyway, the optimal solution was to write php by hand in a Linux vm running on my Note 3. Luckily samsung had a dock thing back then, so I had monitor, keyboard, etc. However, if dex existed then, and I could have just plugged my phone into the computer I used for other work, I absolutely would have. (I'll admit, it was quite fun meeting with the "higher ups" and pulling out my phone for the presentations..)

---

It seems like bVNC is going to see an increase in dex users now that VNC Viewer is no longer supporting dex and UserLAnd opens sessions with it by default. Personally, I hope that, moving forward, bVNC can support dex users, in all capacities. However, I do understand that with limited development resources, only so much can be done.
